### PR TITLE
MM-39632: Migrate from using gorp to sqlx

### PR DESCRIPTION
**Pull request by erni27**
#### Summary

Migrate from using `gorp` to `sqlx` in `store/sqlstore/user_terms_of_service.go`.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18870
JIRA: https://mattermost.atlassian.net/browse/MM-39632 

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/user_terms_of_service.go [hunk 1]
```diff
@@ -32,32 +32,47 @@ func (s SqlUserTermsOfServiceStore) createIndexesIfNotExists() {
...
 		return nil, err
 	}
 
-	c, err := s.GetMaster().Update(userTermsOfService)
+	query := `
+		UPDATE UserTermsOfService
+		SET UserId = :UserId, TermsOfServiceId = :TermsOfServiceId, CreateAt = :CreateAt
+		WHERE UserId = :UserId AND TermsOfServiceId = :TermsOfServiceId
+	`
+	result, err := s.GetMasterX().NamedExec(query, *userTermsOfService)
```
Comment (agnivade) [inline store/sqlstore/user_terms_of_service.go:63]: It's fine to just pass the pointer here.

**Review by agnivade — COMMENTED** [review]
Looking good! Just one small thing.

**Review by agnivade — APPROVED** [review]
Thank you!

**Comment by erni27** [comment]
Seems like e2e tests failed. How can I investigate the issue @agnivade? Guess I don't have acess to the e2e pipleline.

**Comment by agnivade** [comment]
/update-branch

It's ee, not e2e :) The branch was just out of date. I have updated it now.

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks for the contribution @erni27 🎉

**Pull request merged**